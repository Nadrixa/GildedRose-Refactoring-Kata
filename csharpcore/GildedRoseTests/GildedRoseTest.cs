using System;
using Xunit;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using GildedRoseKata;

namespace GildedRoseTests
{
    public class GildedRoseTest
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void ShouldKeepTheSameBehaviour()
        {
            IList<Item> items = new List<Item>
            {
                new() { Name = "foo", SellIn = 0, Quality = 0 },
                new() { Name = "foo", SellIn = 0, Quality = 1 },
                new() { Name = "foo", SellIn = 0, Quality = 2 },
                new() { Name = "Aged Brie", SellIn = 0, Quality = 1 },
                new() { Name = "Backstage passes to a TAFKAL80ETC concert", SellIn = 0, Quality = 1 },
                new() { Name = "Backstage passes to a TAFKAL80ETC concert", SellIn = 1, Quality = 1 },
                new() { Name = "Sulfuras, Hand of Ragnaros", SellIn = 0, Quality = 80 }
            };
            GildedRose app = new GildedRose(items);
            app.UpdateQuality();
            var itemsDetail = from item in items
                select $"{item.Name}, {item.SellIn}, {item.Quality}";
            Approvals.Verify(String.Join("\n", itemsDetail));
        }
    }
}
