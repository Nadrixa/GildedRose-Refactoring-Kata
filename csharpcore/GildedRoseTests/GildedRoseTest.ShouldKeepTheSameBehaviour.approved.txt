foo, -1, 0
foo, -1, 0
foo, -1, 0
Aged Brie, -1, 3
Backstage passes to a TAFKAL80ETC concert, -1, 0
Backstage passes to a TAFKAL80ETC concert, 0, 4
Sulfuras, Hand of Ragnaros, 0, 80
Conjured, -1, 0
Conjured, 0, 0